cmake/clang-format.cmake

```diff
@@ -0,0 +1,83 @@
+set(CLANG-FORMAT_MINIMUM_MAJOR_VERSION 18)
+
+function(get_clang_format_version clang_format_path)
+    set(CLANG_FORMAT_VERSION_OUTPUT "")
+    execute_process(
+        COMMAND ${clang_format_path} --version
+        OUTPUT_VARIABLE CLANG_FORMAT_VERSION_OUTPUT
+    )
+    string(REGEX MATCH "([0-9]+)\\.([0-9]+)\\.([0-9]+)" CLANG_FORMAT_VERSION_OUTPUT ${CLANG_FORMAT_VERSION_OUTPUT})
+    set(CLANG_FORMAT_MAJOR_VERSION ${CMAKE_MATCH_1} PARENT_SCOPE)
+    set(CLANG_FORMAT_MINOR_VERSION ${CMAKE_MATCH_2} PARENT_SCOPE)
+    set(CLANG_FORMAT_PATCH_VERSION ${CMAKE_MATCH_3} PARENT_SCOPE)
+endfunction()
+
+function(check_clang-format_version validator_result_var item)
+    set(${validator_result_var} FALSE PARENT_SCOPE)
+    get_clang_format_version(${item})
+    if (CLANG_FORMAT_MAJOR_VERSION LESS CLANG-FORMAT_MINIMUM_MAJOR_VERSION)
+        message(DEBUG "clang-format found at ${item} | version: ${CLANG_FORMAT_MAJOR_VERSION}.${CLANG_FORMAT_MINOR_VERSION}.${CLANG_FORMAT_PATCH_VERSION}")
+        message(DEBUG "but version is lower than ${CLANG-FORMAT_MINIMUM_MAJOR_VERSION}")
+        set(${validator_result_var} FALSE PARENT_SCOPE)
+    else()
+        set(${validator_result_var} TRUE PARENT_SCOPE)
+    endif()
+endfunction()
+
+function(print_clang_format_install_instructions)
+    message(STATUS "🛠️ Please install clang-format to enable code formatting")
```

Comment:
Need not be addressed: we could mention that [`clang-format` is available for most platform on conda-forge](https://prefix.dev/channels/conda-forge/packages/clang-format).